core/base/CMakeLists.txt

```
@@ -198,11 +198,27 @@ set(BASE_SOURCES
   src/TVirtualX.cxx
 )
 
+set(FULL_BASE_HEADERS ${BASE_HEADERS})
+list(TRANSFORM FULL_BASE_HEADERS PREPEND ${CMAKE_CURRENT_SOURCE_DIR}/inc/)
+
 # only here complete list of headers can be propogated to parent cmake file
 set_property(TARGET Core APPEND PROPERTY DICT_HEADERS ${BASE_HEADERS})
 
 target_sources(Core PRIVATE ${BASE_SOURCES})
 
+if(NOT CMAKE_VERSION VERSION_LESS "3.23.0") # https://discourse.cmake.org/t/file-set-xyz-is-listed-in-interface-file-sets-of-w-but-has-not-been-exported/9131/3
+  target_sources(
+      Core
+      PRIVATE
+      FILE_SET private_header_files
+      TYPE HEADERS
+      BASE_DIRS inc/ src/
+      FILES
+          ${CMAKE_CURRENT_SOURCE_DIR}/src/TListOfTypes.h
+          ${FULL_BASE_HEADERS}
```

BASE_DIRS should make the headers findable, and there's nothing in `src` that needs to be picked up using a relative path, I believe:
```suggestion
      BASE_DIRS inc/
      FILES
          ${BASE_HEADERS}
          ${CMAKE_CURRENT_SOURCE_DIR}/src/TListOfTypes.h
```

---

> there's nothing in `src` that needs to be picked up

Btw, we could split things into two file-sets, one called private_headers and the other public_headers.

Later on, we would only "INSTALL" the public ones I guess, using sth like:
https://discourse.cmake.org/t/cmake-file-set-for-header-when-using-destination/5692/2

Here is a full example of 'migration' using also a VERIFY target. https://github.com/bemanproject/exemplar/pull/30/files

In principle, we could then also get rid of the copy-headers-to-build-dir, but that's orthogonal to this PR. This is an independent first step that makes the IDE work better.
That's the reason that I did not remove (yet) the `target_include_directories(build_dir/inc)`